Rename "Downloads" to "Install" and redesign

jvns opened the pull request:
This renames "Downloads" to "Install" and redesigns the page to simplify it a lot. You can see it in action here: https://jvns.github.io/git-scm.com/install/linux

I'm still not sure where the Logos page should be linked. Maybe in About?

I've designed it as "tabs", where each tab is a separate page. I did it that way for accessibility (though I'm very far from an accessibility expert): the idea is that it would be easier for someone using a screenreader if they don't have to navigate through a bunch of irrelevant content. The `install-header` shortcode is rendered in an awkward way (you have to close some <div> tags that are secretly updated in the tab header), but I didn't think of a better way to do it yet.

The Install links on the sidebar and homepage get automatically updated with Javascript to point to the appropriate page.

The Build from Source page is pretty sparse right now, maybe it would be good in the future to point users to instructions for how to build Git from source.

Todos:
- [x] Figure out Logos section
- [x] Fix tabs in mobile version somehow (right now they don't all fit)
- [x] Double check that all aliases are updated correctly
- [x] Maybe rename /install/win to /install/windows?

jvns committed:
Move /download to /install

- Add a new style where we display the install instructions as tabs,
  with a shortcode and CSS
- Update aliases
- Move existing Windows/Linux/Mac pages to the new style, keeping the
  content the same

Signed-off-by: Julia Evans <julia@jvns.ca>

jvns committed:
install page: Rename 'downloads" to "install" in sidebar and homepage

Signed-off-by: Julia Evans <julia@jvns.ca>

jvns committed:
install page: Automatically link to correct OS directions

Add a script that will update the homepage / sidebar links to the
appropriate OS-specific link.

This falls back gracefully: if the script doesn't run, you can just
go to /install/ and click on the appropriate page.

Signed-off-by: Julia Evans <julia@jvns.ca>

jvns committed:
install page: add new "Build from source" page

Signed-off-by: Julia Evans <julia@jvns.ca>

jvns committed:
install page: remove "build from source" note on Linux/Mac

We already have a "Build from source" page

dscho commented on layouts/partials/sidebar.html:
```diff
@@ -54,12 +54,7 @@
-            <a href="{{ relURL "downloads/logos" }}"{{ if (eq .Params.Subsection "logos") }} class="active"{{ end }}>Logos</a>
-          </li>
-        </ul>
+        <a href="{{ relURL "install" }}"{{ if (eq $section "install") }} class="active"{{ end }}>Install</a>
```
I think this is the reason why the Playwright tests [fail](https://github.com/git/git-scm.com/pull/2096/files#annotation_39710187708) now...

```
Error: expect.not.toHaveClass: Error: strict mode violation: getByRole('link', { name: 'All' }) resolved to 2 elements:
    1) <a class="install-link" href="http://localhost:5000/install/win">Install</a> aka getByRole('link', { name: 'Install' })
    2) <a href="#" data-os="" class="subtle-button gui-os-filter">All</a> aka getByRole('link', { name: 'All', exact: true })

Call log:
  - Expect "not toHaveClass" with timeout 5000ms
  - waiting for getByRole('link', { name: 'All' })


  66 |
  67 |     const allButton = page.getByRole('link', { name: 'All' })
> 68 |     await expect(allButton).not.toHaveClass(/selected/)
     |                                 ^
  69 |
  70 |     const thumbnails = page.locator('.gui-thumbnails li:visible')
  71 |     const count = await thumbnails.count()
    at /home/runner/work/git-scm.com/git-scm.com/tests/git-scm.spec.js:68:33
```

We probably need to change [this line](https://github.com/git/git-scm.com/blob/e2eb96fa7bc9f7c883fade034efc550df442d59e/tests/git-scm.spec.js#L67):

```js
    const allButton = page.getByRole('link', { name: 'All' })
```

to [use an exact match](https://playwright.dev/docs/api/class-locator#locator-get-by-role-option-exact):

```js
    const allButton = page.getByRole('link', { name: 'All', exact: true })
```

jvns replied:
thank you for looking into that, will try that fix!

dscho reviewed (approved):
This looks really good!

jvns committed:
install page: rename /install/win to /install/windows

jvns committed:
Move /downloads/logos to /community/logos

We're renaming the download page to /install, so the logos page has
to go somewhere. Reasons to move it to /community:

- That's what Python does https://www.python.org/community/logos/
- I think the About section needs to be reworked, so it may not exist
  in the future

jvns committed:
Try to fix failing Playwright test

jvns commented:
Okay, I've fixed the mobile issues and I think this is ready to go now if:

a) the tests pass, and 
b) folks think that moving the logos to the "community" section makes sense

I moved it to the "Community" section because that's what [Python does](https://www.python.org/community/logos/), though it might be nice at some point to be clearer about whether folks are encouraged to use the logos in their projects (linking to the trademark policy as necessary).

You can see it here https://jvns.github.io/git-scm.com/community/logos

dscho commented:
> a) the tests pass, and

They pass!

> b) folks think that moving the logos to the "community" section makes sense

I am just one "folk" among many, but you have my vote that it makes sense 😃

jvns committed:
install page: fix links on homepage

jvns commented:
(just fixed a few stray links to the old page I noticed)
